src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -66,11 +66,23 @@ public IStringLocalizer Create(Type resourceSource)
             var typeInfo = resourceSource.GetTypeInfo();
             var assembly = typeInfo.Assembly;
 
-            // Re-root the base name if a resources path is set
-            var baseName = string.IsNullOrEmpty(_resourcesRelativePath)
-                ? typeInfo.FullName
-                : _hostingEnvironment.ApplicationName + "." + _resourcesRelativePath
-                    + TrimPrefix(typeInfo.FullName, _hostingEnvironment.ApplicationName + ".");
+            string baseName;
+
+            if (_hostingEnvironment.ApplicationName == null)
+            {
+                var assemblyName = assembly.GetName().Name + ".";
```

Comment:
I would name that assemblyPathPrefix or something like that. When you say assemblyName I think of A.B.C or SomeAssembly, not something that ends with a dot and that makes my brain tweak when I see you concatenating paths below. (I had to go back to this part of the code, that's why I'm pointing it out).